python/cuspatial/cuspatial/tests/binpreds/binpred_test_dispatch.py

```diff
@@ -505,7 +505,43 @@ def object_dispatch(name_list):
 
 
 def simple_test_dispatch():
-    """Generates a list of test cases for each geometry type combination."""
+    """Generates a list of test cases for each geometry type combination.
+
+    Each dispatched test case is a tuple of the form:
+    (test_name, test_description, lhs, rhs)
+    which is run in `test_binpred_test_dispatch.py`.
+
+    The test_name is a unique identifier for the test case.
+    The test_description is a string representation of the test case.
+    The lhs and rhs are GeoSeries of the left and right geometries.
+
+    lhs and rhs are always constructed as a list of 3 geometries since
+    the binpred function is designed to operate primarily on groups of
+    geometries. The first and third feature in the list always match
+    the first geometry specified in `test_description`, and the rhs is always
+    a group of three of the second geometry specified in `test_description`.
+    The second feature in the lhs varies.
+
+    When the types of the lhs and rhs are equal, the second geometry
+    from `test_description` is substituted for the second geometry in the lhs.
+    This produces a test form of:
+    lhs     rhs
+    A       B
+    B       B
+    A       B
```

Comment:
For even greater variation, would it make sense to include `B.op(A)` in this series?

Reply:
Good idea, this important detail is covered in #1064 where every feature is tested for `lhs.predicate(rhs)` and also `rhs.predicate(lhs)`.